lib/storage/index_db.go

```diff
@@ -2210,6 +2210,12 @@ func (is *indexSearch) searchMetricIDs(qt *querytracer.Tracer, tfss []*TagFilter
 	return sortedMetricIDs, nil
 }
 
+func errTooManyTimeseries(maxMetrics int) error {
+	return fmt.Errorf("the number of matching timeseries exceeds %d; "+
+		"either narrow down the search or increase -search.max* command-line flag values at vmselect; "+
```

Comment:
Side note: this error message isn't very helpful for users, since it doesn't point to the concrete limit which was exceeded. The provided url contains information about big number of different limits, and it is hard to figure out which limit was exceeded from this url. In most cases this error means that the `-searchaxUniqueTimeseries` limit is exceeded. How about mentioning this limit in the error message in the following way:

```
either narrow down the search or increase -search.max* command-line flag values at vmselect (the most likely limit is -search.maxUniqueTimeseries); see ...
```

This change can be made either in this pull request or in the follow-up pull request for the sake of more clear commit history.

Reply:
Ack. Will create a follow-up PR.